lib/system/linux/device.c

```diff
@@ -447,10 +447,8 @@ static int metal_linux_dev_open(struct metal_bus *bus,
 
 		/* Try and open the device. */
 		error = ldrv->dev_open(lbus, ldev);
-		if (error) {
-			ldrv->dev_close(lbus, ldev);
+		if (error)
 			continue;
```

Comment:
Better to leave it, as it forces the driver dependent device open to clean up properly when it has issues during opening.
E.g. the uio_dev_open() didn't do that.
Or does it introduce any issues?

Reply:
If open fails, that means the device isn't opened. Closing such device doesn't seem straightforward. As an example, posix open/close can be looked as analogy. If any cleanup is needed, better to be done in the error handling in the open call.
In my case, the close is calling metal_irq_unregister() with invalid fd, and it was printing error, even when everything is still operational. So I removed this to avoid misleading.

Reply:
ok, I see, better to have another patch to the existing driver dev_open to make sure it clean up when it fails in the dev_open(). But this can be another pull request.
Please go ahead with your change for this pull request. We can worry about to cleanup properly in the driver's dev_open() later